python/lsst/qa/explorer/functors.py

```diff
@@ -767,3 +793,169 @@ class Seeing(Functor):
     def _func(self, df):
         return 0.168*2.35*np.sqrt(0.5*(df['ext_shapeHSM_HsmPsfMoments_xx']**2 +
                                        df['ext_shapeHSM_HsmPsfMoments_yy']**2))
+
+
+class e1(Functor):
+    name = "Ellipticity e1"
+
+    def __init__(self, colXX, colXY, colYY, **kwargs):
+        self.colXX = colXX
+        self.colXY = colXY
+        self.colYY = colYY
+        self._columns = [self.colXX, self.colXY, self.colYY]
+        super().__init__(**kwargs)
+
+    @property
+    def columns(self):
+        return [self.colXX, self.colXY, self.colYY]
+
+    def _func(self, df):
+        return df[self.colXX] - df[self.colYY] / (df[self.colXX] + df[self.colYY])
+
+
+class e2(Functor):
+    name = "Ellipticity e2"
+
+    def __init__(self, colXX, colXY, colYY, **kwargs):
+        self.colXX = colXX
+        self.colXY = colXY
+        self.colYY = colYY
+        super().__init__(**kwargs)
+
+    @property
+    def columns(self):
+        return [self.colXX, self.colXY, self.colYY]
+
+    def _func(self, df):
+        return 2*df[self.colXY] / (df[self.colXX] + df[self.colYY])
+
+
+class EffectiveRadius(Functor):
+
+    def __init__(self, colXX, colXY, colYY, **kwargs):
+        self.colXX = colXX
+        self.colXY = colXY
+        self.colYY = colYY
+        super().__init__(**kwargs)
+
+    def computeReScalar(xx, yy, xy):
+        quad = afwGeom.Quadrupole(xx, yy, xy)
+        axes = afwGeom.ellipses.Axes(quad)
+        return np.sqrt(axes.getA() * axes.getB())
+
+    computeReVec = np.vectorize(computeReScalar)
+
+    @property
+    def columns(self):
+        return [self.colXX, self.colXY, self.colYY]
+
+    def _func(self, df):
+        reArr = self.computeReVec(df[self.colXX], df[self.colYY], df[self.colXY])
```

Comment:
Faster vectorization of this is:
```
rArr = (df[self.colXX]*df[self.colYY] - df[self.colXY]**2)**0.25
```
(i.e. if XX, YY, XY are the elements of a symmetric 2x2 matrix, `rArr` is the 4th root of its determinant...hence "determinant radius" in `afw.geom.ellipses`).